src/neo4j_genai/retrievers/__init__.py

```diff
@@ -12,3 +12,34 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
+
+from .hybrid import HybridCypherRetriever, HybridRetriever
+from .text2cypher import Text2CypherRetriever
+from .vector import VectorCypherRetriever, VectorRetriever
+import importlib
+
+__all__ = [
+    "VectorRetriever",
+    "VectorCypherRetriever",
+    "HybridRetriever",
+    "HybridCypherRetriever",
+    "Text2CypherRetriever",
+]
+
+
+pinecone_spec = importlib.util.find_spec("pinecone")
+if pinecone_spec:
```

Comment:
With this `if`, users that do not have `pinecone` installed won't even be able import the `PineconeRetriever` from `neo4j_genai.retrievers`, right?

Reply:
There is another way, which I think is clearer for users since it will raise a meaningful error, see for instance how we deal with `openai` package here: https://github.com/neo4j/neo4j-genai-python/blob/73a05fd58cedbd961416ead2dc57b86726617639/src/neo4j_genai/llm/openai_llm.py 
The exception is raised only when you try to instantiate the class.
Any toughts?

Reply:
When I try importing pinecone or weaviate in the __init__.py, I get a recommendation from ruff to use `importlib.util.find_spec` instead.

I'll go with your suggestion though and maybe add a flag for ruff to ignore, I think it's clearer to the user.

Reply:
[Added the change here ](https://github.com/neo4j/neo4j-genai-python/pull/63/commits/37ca04c6abae96c3a37d6de7f111b2334aab5a85#diff-d94b6d8b65a95a5451fe005a015a7c310c36925af9816d70b1f798011baaf8f5L30)